authLdap.php

```diff
@@ -362,13 +360,17 @@ function authLdap_login($user, $username, $password, $already_md5 = false)
                 $user_info['user_email'] = $username . '@example.com';
             }
             $userid = wp_insert_user($user_info);
+            if (is_wp_error($userid)) {
+                authLdap_debug('Error creating user : ' . $userid->get_error_message());
+                trigger_error('Error creating user: ' . $userid->get_error_message());
+                return $userid;
+            }
         }
 
-        // if the user exists, wp_insert_user will update the existing user record
-        if (is_wp_error($userid)) {
-            authLdap_debug('Error creating user : ' . $userid->get_error_message());
-            trigger_error('Error creating user: ' . $userid->get_error_message());
-            return $userid;
```

Comment:
This also caught issues when updating a user. Therefore I'd rather leave it at this place!